program-runtime/src/loaded_programs.rs

```diff
@@ -102,6 +102,31 @@ impl From<ProgramCacheEntryOwner> for Pubkey {
     }
 }
 
+/*
+    The possible ProgramCacheEntryType transitions:
+
+    Builtin re/deployment:
+    - Empty => Builtin in TransactionBatchProcessor::add_builtin
+    - Builtin => Builtin in TransactionBatchProcessor::add_builtin
+
+    Un/re/deployment (with delay and cooldown):
+    - Empty / Closed => Loaded+DelayVisibility in UpgradeableLoaderInstruction::DeployWithMaxDataLen
+    - Loaded / FailedVerification => Loaded+DelayVisibility in UpgradeableLoaderInstruction::Upgrade
+    - Loaded / FailedVerification => Closed in UpgradeableLoaderInstruction::Close
+    - Loaded+DelayVisibility => Loaded in ProgramCacheForTxBatch::find
```

Comment:
Can a program with `DelayVisibility` transition to `Loaded` through `find()`? My understanding is that `find()` should actually treat the program as missing/delay_visible.

Reply:
Yes that is what I tried to express with `Loaded+DelayVisibility`, but obviously needs a better expression.